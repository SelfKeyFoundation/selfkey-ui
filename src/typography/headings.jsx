import * as React from 'react';
import { css, withStyles } from '../default/with-styles';


class H1 extends React.Component {
    constructor(props) {
      super(props);
    }
  
    render() {
        const {
            styles,
            } = this.props;
        return (
            <h1 {...css(styles.Header)}>
                {this.props.children}
            </h1>
        );
    }
  }

export default withStyles(({ selfkeyWallet: { color } }) => ({
    Header: {
        color: color.core.white,
        fontFamily: '"OrbitronMedium"',
        fontWeight: 500,
        textTransform: 'uppercase',
        marginTop: '24px',
        letterSpacing: '4px',
        fontSize: '26px'
    },


}))(H1);

