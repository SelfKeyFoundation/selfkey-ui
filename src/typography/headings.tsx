import * as React from 'react';
// @ts-ignore
import injectSheet, { StyleSheet, StyledComponentProps } from 'react-jss';

const styles: StyleSheet = {
    base: {
        color: '#fff',
        fontFamily: 'ProximaNovaRegular, arial, sans-serif',
        fontWeight: 400
    },

    h1: {
        fontSize: '26px'
    },

    h2: {
        fontSize: '22px'
    },
    
    h3: {
        fontSize: '18px'
    }
};

export type HeadingsProps = {
    className?: string
}

export const H1 = injectSheet(styles)<HeadingsProps>(({ classes, children, className }) => (
    <h1 className={`${classes.base} ${classes.h1} ${className}`}>{children}</h1>
));


export const H2 = injectSheet(styles)<HeadingsProps>(({ classes, children, className}) => (
    <h2 className={`${classes.base} ${classes.h2} ${className}`}>{children}</h2>
));


export const H3 = injectSheet(styles)<HeadingsProps>(({ classes, children, className}) => (
    <h3 className={`${classes.base} ${classes.h3} ${className}`}>{children}</h3>
));

export default H1;
