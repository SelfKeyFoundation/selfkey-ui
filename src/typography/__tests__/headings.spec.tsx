import * as React from 'react';
import { create } from 'react-test-renderer';
import { H1, H2, H3 } from '../headings';

describe('Headings', () => {
  it('should show Heading 1', () => {
    const component = create(
      <H1>TEST</H1>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show Heading 2', () => {
    const component = create(
      <H2>TEST</H2>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show Heading 3', () => {
    const component = create(
      <H3>TEST</H3>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});





