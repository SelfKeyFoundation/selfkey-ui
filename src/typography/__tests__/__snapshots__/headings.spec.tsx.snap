// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headings should show Heading 1 1`] = `
<h1
  className="Component-base-0-1-1 Component-h1-0-1-2 undefined"
>
  TEST
</h1>
`;

exports[`Headings should show Heading 2 1`] = `
<h1
  className="Component-base-0-1-5 Component-h2-0-1-7 undefined"
>
  TEST
</h1>
`;
